### PR TITLE
Adding CHANGELOG and improve release process

### PR DESCRIPTION
**Pull request by eduardogr**
- Changelog
  - Adding 3.6.3 release notes to CHANGELOG.md
- Release
  - Adding release job in circleci without actually releasing bittensor (dry run to see if everything looks good)
  - Improving release script to be able to execute a dry run with no secrets
  - Enabling circleci to build docker image
  - Reorganizing circleci workflows
  - Update release documentation and scripts

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -137,10 +179,33 @@ workflows:
...
+            branches:
+              only:
+                - /^(release|hotfix)/.*/
+      - release-dry-run:
+          filters:
+            branches:
+              only:
+                - /^(release|hotfix)/.*/
+  
+  release-requirements:
```
Comment (eduardogr) [inline .circleci/config.yml:200]: when something reach the default branch

File: .circleci/config.yml [hunk 2]
```diff
@@ -137,10 +179,33 @@ workflows:
           #requires:
             #- build-and-test
 
-  check-version-updated:
+  release-branches-requirements:
```
Comment (eduardogr) [inline .circleci/config.yml:182]: when creating expected release branches:
  - release/*
  - hotfix/*

**Review by camfairchild — APPROVED** [review]
Looks great :)

File: scripts/release/add_notes_changelog.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
```
Comment (eduardogr) [inline scripts/release/add_notes_changelog.sh:1]: right now I expect the ones creating release branches to execute this script @Eugene-hu 

I've added this to the [RELEASE_GUIDELINES.md](https://github.com/opentensor/bittensor/pull/1082/files#diff-b3fd13d65d94ab4865ee5875e9932171928a4a3da0bf3303e72364f6c74015a0) document

**Review by Eugene-hu — APPROVED** [review]
LGTM

File: scripts/release/add_notes_changelog.sh (hunk 3 above)
Comment (Eugene-hu) [reply scripts/release/add_notes_changelog.sh:1]: Sounds good! Will give it a try for the next release!

**Pull request merged**